### PR TITLE
Update publish dev pypi script to use cirq-dev

### PR DESCRIPTION
**Pull request by dabacon**
I'm keeping the set-dev-version script in the case that we can go back to using travis's deploy.

### CODE REVIEW

**Comment by dabacon** [comment]
@Strilanc if you look at this I can do a follow up turning back on dev deploys.

File: dev_tools/packaging/publish-dev-package.sh [hunk 1]
```diff
@@ -105,6 +107,9 @@ cd "$(git rev-parse --show-toplevel)"
 tmp_package_dir=$(mktemp -d "/tmp/publish-dev-package_package.XXXXXXXXXXXXXXXX")
 trap "{ rm -rf ${tmp_package_dir}; }" EXIT
 
+# Configure to push to cirq-dev and not cirq.
+export CIRQ_DEV_VERSION=`dev_tools/packaging/set-dev-version.sh`
```
Comment (Strilanc) [inline dev_tools/packaging/publish-dev-package.sh]: `export CIRQ_DEV_VERSION=$(dev_tools/packaging/set-dev-version.sh)` to be consistent with the rest of the file

File: dev_tools/packaging/set-dev-version.sh [hunk 2]
```diff
@@ -19,49 +19,32 @@
 # the CIRQ_DEV_VERSION variable to a string that is the dev version
```
Comment (Strilanc) [inline dev_tools/packaging/set-dev-version.sh]: Update this description to say that the script prints a new dev version. Also rename the script to `pick-dev-version` or `generate-dev-version-id` or similar to make this clearer.

File: dev_tools/packaging/publish-dev-package.sh (hunk 1 above)
Comment (dabacon) [reply dev_tools/packaging/publish-dev-package.sh]: done

File: dev_tools/packaging/set-dev-version.sh (hunk 2 above)
Comment (dabacon) [reply dev_tools/packaging/set-dev-version.sh]: done

File: dev_tools/packaging/generate-dev-version-id.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ACTUAL_VERSION=`echo $ACTUAL_VERSION_LINE | cut -d'"' -f 2`
+
+if [[ ${ACTUAL_VERSION} == *"dev" ]]; then
+  echo "${ACTUAL_VERSION}$(date "+%Y%m%d%H%M%S")"
+else
+  echo "Version doesn't end in dev: ${ACTUAL_VERSION_LINE}" >&2
+  exit 1
+fi
+
+exit 0
```
Comment (Strilanc) [inline dev_tools/packaging/generate-dev-version-id.sh]: `exit 0` is redundant at the end of a script.

Comment (dabacon) [reply dev_tools/packaging/generate-dev-version-id.sh]: done

**Pull request merged**